docs/project-file-structure.md

```diff
@@ -64,30 +91,139 @@ to organize their structure
 
 Resources, scenes, scripts etc. used exclusively within that scene should be stored in the scenes subdirectory
 
+A scene and its script should always be together within the directory i.e. no /scripts/ and /scenes/ directories should exist within
+the project somewhere.
+
 Example:
 ```
 root/
 |- src/
-    |- Theme/
+    |- theme/
     |   |- ButtonNormal.tres
     |   |- ButtonHover.tres
     |   |- ButtonDisabled.tres
-    |- Scripts/
+    |- utility/
     |   |- Math.gd
-    |- Menu/
+    |- menu/
     |   |- MenuManager.tscn
     |   |- MenuManager.gd
     |   |- MainMenu/
     |   |   |- MainMenu.tscn
     |   |   |- MainMenu.gd
     |   |   |- MainMenuFont.ttf
     |   |   |- MainMenuLabel.tres
-    |   |- SettingsMenu/
+    |   |- settings_menu/
     |   |   |- Settings.tscn
     |   |   |- Settings.gd
     |   |   |- Resolution/
     |   |   |   |- ResolutionSelector.gd
-    |   |   |- Volume/
+    |   |   |- volume/
     |   |   |   |- VolumeGrid.tscn
     |   |   |   |- VolumeGrid.gd
 ```
+
+#### Naming Convention
+
+Directory names should be snake_case.
+```
+root/src/game_session/
+```
+
+File names should be PascalCase.
+```
+root/src/game_session/Map.tscn
+```
+
+### Asset Organization
+
+Note: Asset files should not be exported with Godot in a pck file. They should be downloaded via an installer or handled in a similar
+fashion to the data files and their exclusion from the export
+
+Asset files should be organized within a relevant category from root/
+```
+root/art/
+root/music/
+root/sfx/
+```
+
+Asset files should be organized in a subdirectory that specifies it's relevancy within the application.
+```
+root/art/economy/
+root/art/military/
+root/sfx/economy/
+root/sfx/technology/
+```
+
+Asset files should be prefixed by the asset type
+```
+iconCoal.png
+modelSoldier.obj
+ostTheCrown.mp3
+videoIntro.mp4
+```
+
+The prefix can be removed if the file is within a subdirectory that replaces the asset type
+```
+./iconCoal.png -> ./icon/Coal.png
+./ostTheCrown.png -> ./ost/TheCrown.mp3
+```
+
+#### Naming Convention
+
+Directories should be snake_case.
+
+Asset files should be PascalCase for their name.
+```
+TheCrown.mp3
+Coal.png
+```
+
+Asset files with a prefix should have that prefix be lowercase.
+```
+ostTheCrown.mp3
+iconCoal.png
+```
+
+### Data File Organization
+
+All data files should be in the root/common/ directory.
+
+Data files should be organized by category.
+```
+root/common/map/
+root/common/technology/
+root/common/pop/
+```
+
+Data files that are plain text should have the ov2d file extension.
+```
+root/common/pop/soldier.ov2d
+```
+
+Data files that are complex data like a bitmap should make use of the standard file extension for that data
+```
+root/common/map/provinces.bmp
+```
+
+Data files can be further organized by subdirectory or prefix.
+```
+root/common/economy/goods/Coal.ov2d
+root/common/economy/goodsCoal.ov2d
+```
```

Comment:
I'll keep the prefix discussion here because it is first for me but it applies for all suggestions.

The use of prefixes allows us to identify what a file is used for even when it is located in a non-specific directory eg. root/art/iconGeneral1.png vs root/art/General1.png

Sub-directories also make it difficult to navigate through the project, especially when the given sub-directory only contains one file. This is not necessarily relevant for the core game as it is very unlikely we will have just one asset for a given type, but it is a qol feature that might be useful within modding.

Reply:
I don't see how a subdirectory makes the projekt more difficult to navigate tbh, if it needs a prefix, it needs a subdirectory imo. With modding I'm not sure how adding expectations to prefix stuff is less confusing/difficult then just seeing all good icons go in the r/art/goods folder.

Reply:
Fair enough, I personally like to keep my directories on the surface level as much as possible so I can quickly jump back and forth between them.

Prefixes do the exact same thing as a sub-directory, but only having one or two files in a sub-directory doesn't make sense to me (1st paragraph).

Reply:
To paraphrase a certain pirate-related movie: "It's more what you'd call guidelines than actual rules"

I think this is enough of an edge-case that we can re-evaluate/address it if we start seeing lots of directories with only a couple of assets